### PR TITLE
Remove coffee script

### PR DESCRIPTION
**Pull request by segayuu**

### CODE REVIEW

**Comment by yoshinorin** [comment]
I will review & test it this weekend if I have a time :)

File: gulp-task/cli.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+module.exports = (profile = "default") => {
+  const cfg = config["cli:" + profile];
+  const mappings = cfg.opts.mapping;
+  const globalMapping = mappings.GLOBAL || {};
+  const getValue = (o, k) => {
```
Comment (yoshinorin) [inline gulp-task/cli.js]: It seems unused. Is this anonymous function really needed?

File: gulp-task/cli.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    switch (t) {
+      case 'string':
+        return v;
+      case 'number':
+        return parseFloat(v);
+      case 'boolean':
+        return v.toLowerCase() === 'true';
+      case 'object':
+        if (Array.isArray(orig)) {
+          return v.split(',').map(function(v) {
```
Comment (yoshinorin) [inline gulp-task/cli.js]: You can use allow function here. `.map((v) => {`

File: gulp-task/cli.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      if (value == null) {
+        continue;
+      }
+      // Set value
+      console.log(`Map ${key}.${field} -> --${arg}=${value}`);
+      setValue(target, field, value);
+    }
+  }
+  const results = [];
+  for (const field in globalMapping) {
```
Comment (yoshinorin) [inline gulp-task/cli.js:77]: What is the purpose of this expression?

File: test/content.spec.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  describe('helper', () => {
+    beforeEach(() => sinon.stub(inject, 'raw'));
+
+    afterEach(() => inject.raw.restore());
+
+    const should_generate_raw_html = expected => {
+      inject.raw.calledOnce.should.be.true;
+      inject.raw.calledWith('test').should.be.true;
+      
+      const [x, html, opts] = inject.raw.getCall(0).args;
```
Comment (yoshinorin) [inline test/content.spec.js]: `x` is unused.

**Comment by segayuu** [comment]
If you use `gulp-heap`, node v12 will generate an error (for unknown reasons).
`gulp-heap` is an experimental library, and it can't but support node v12, but gulp's cli task depends on it and it is difficult to maintain.
IMO that the code around gulp is broken.

**Comment by yoshinorin** [comment]
@segayuu 
OK. I may be understood. So, IMHO we should remove `node` from `.travis.yaml` temporary.

Btw, did you already fixed my code review points? If yes, I will review again.

**Comment by segayuu** [comment]
@YoshinoriN  Since there was a lack of comments, I will add it. This error is similar to gulp in v3 and has resolved similar issues gulpjs/gulp#2280 (you can see similar problems from the error message, but I can not understand why). The unmaintained `gulp-heap` can not handle that and is not likely to be usable on node 12.

~~As there was one point missing review point.~~

**Comment by curbengh** [comment]
Why don't we drop gulp and babel altogether? Also drop `src/*.es6` in favor of `lib/*.js`?

**Comment by segayuu** [comment]
@curbengh Changes made in one PR should be made at a verifiable granularity.
I think that gulp, babel and coffeescript are beyond the manageable granularity just by removing each with PR alone.
In the case of multiple overlapping, the burden on the reviewer is too high, so this PR only removes the coffeescript.

**Comment by curbengh** [comment]
@YoshinoriN 
your comments seem to be addressed, can you review this again?
I think the node 12 issue can be ignored for now, it can be fixed by dropping gulp (in a separate PR).
I did [submit a PR](https://github.com/akfish/gulp-heap/pull/1) on gulp-heap that could solve the issue, but the repo seems abandoned.

**Review by yoshinorin — APPROVED** [review]

**Comment by yoshinorin** [comment]
Let's move forward :)

**Comment by segayuu** [comment]
Thanks @YoshinoriN  && @curbengh !

**Pull request merged**